flax/nnx/extract.py

```diff
@@ -343,7 +345,9 @@ def from_tree(
     with graph.merge_context(is_inner, ctxtag) as merge_ctx:
       return jax.tree.map(
         lambda x: merge_fn(merge_ctx, (), prefix, x)
-        if map_non_graph_nodes or is_node_leaf(x)
+        if map_non_graph_nodes
```

Comment:
This lambda becomes kinda long & unreadable. Can you refactor it out to a function? Then you can also use this for the `merge_fn` call in line 374.